.github/workflows/reusable_dockerfile_pipeline.yml

```diff
@@ -35,25 +36,39 @@ jobs:
       - name: Add vars to ENV
         id: setting_env
         run: |
-          echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
-          echo "short_sha=`echo ${GITHUB_SHA} | cut -c1-8`" >> "$GITHUB_OUTPUT"
-          # yamllint disable
-          echo "IMAGE_NAME=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
-          echo "image_name=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')" >> "$GITHUB_OUTPUT"
-          # here we validate if we have specified a different package name in
-          # the inputs, if so, we change the package to it.
-          if [[ ${{ inputs.packageName }} != ${{ github.repository}} ]];then
+          # Extract both the repository owner and repository name
+            REPO_OWNER=$(echo "${{ github.repository }}" | awk -F'/' '{print $1}' | tr '[:upper:]' '[:lower:]')
+            REPO_NAME=$(echo "${{ github.repository }}" | awk -F'/' '{print $2}' | tr '[:upper:]' '[:lower:]')
+
+            echo "Repository Owner: $REPO_OWNER"
+            echo "Repository Name: $REPO_NAME"
+
+            echo "REPO_OWNER=$REPO_OWNER" >> $GITHUB_ENV
+            echo "repo_owner=$REPO_OWNER" >> "$GITHUB_OUTPUT"
+
+            echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
+            echo "short_sha=`echo ${GITHUB_SHA} | cut -c1-8`" >> "$GITHUB_OUTPUT"
+            # yamllint disable
+
+            echo "IMAGE_NAME=$REPO_NAME" >> $GITHUB_ENV
+            echo "image_name=$REPO_NAME" >> "$GITHUB_OUTPUT"
+
+            # here we validate if we have specified a different package name in
+            # the inputs, if so, we change the package to it.
+            if [[ ${{ inputs.packageName }} != ${{ github.repository}} ]];then
             # validate the input package name characters
-            if [[ ! "${{ inputs.packageName }}" =~ ^[A-Za-z0-9\-]+$ ]]; then
-              echo "------------------------------------------------------------"
-              echo "ERROR: Package name not valid! => [ ${{ inputs.packageName }} ]"
-              echo "ONLY can use: A-Za-z0-9\-"
-              echo "------------------------------------------------------------"
-              exit 1
+              if [[ ! "${{ inputs.packageName }}" =~ ^[A-Za-z0-9\-]+$ ]]; then
+                echo "------------------------------------------------------------"
+                echo "ERROR: Package name not valid! => [ ${{ inputs.packageName }} ]"
+                echo "ONLY can use: A-Za-z0-9\-"
+                echo "------------------------------------------------------------"
+                exit 1
+              fi
+              PACKAGE_NAME=$(echo "${{ inputs.packageName }}" | tr '[:upper:]' '[:lower:]')
+              echo "IMAGE_NAME=$PACKAGE_NAME" >> $GITHUB_ENV
+              echo "image_name=$PACKAGE_NAME" >> "$GITHUB_OUTPUT"
+
             fi
```

Comment:
I think we can clean this logic up and I think we are missing a potential bug of someone naming a repository something that is an invalid package name.

```
              # Set the default package name to the repository name
              PACKAGE_NAME=$REPO_NAME

              # If there is a user submitted package name, use it
              if [[ ! "${{ inputs.packageName }}" != "" ]]; then
                PACKAGE_NAME=$(echo "${{ inputs.packageName }}" | tr '[:upper:]' '[:lower:]')
              fi

              # validate the package name characters
              if [[ ! $PACKAGE_NAME =~ ^[A-Za-z0-9\-]+$ ]]; then
                echo "------------------------------------------------------------"
                echo "ERROR: Package name not valid! => [ $PACKAGE_NAME} ]"
                echo "ONLY can use: A-Za-z0-9\-"
                echo "------------------------------------------------------------"
                exit 1
              fi

              echo "IMAGE_NAME=$PACKAGE_NAME" >> $GITHUB_ENV
              echo "image_name=$PACKAGE_NAME" >> "$GITHUB_OUTPUT"
            fi
```

Reply:
Done